lib/packages/docutils/rst.nim

```diff
@@ -578,35 +578,41 @@ proc whichRoleAux(sym: string): RstNodeKind =
   else:  # unknown role
     result = rnUnknownRole
 
-proc newSharedState(options: RstParseOptions,
-                    findFile: FindFileHandler,
-                    msgHandler: MsgHandler): PSharedState =
+proc newSharedStateRst*(options: RstParseOptions,
+                        filename: string,
+                        findFile: FindFileHandler,
+                        msgHandler: MsgHandler): RstSharedState =
   new(result)
```

Comment:
in this or future cleanup PR:
result = RstSharedState(a1: a1, a2: a2, ...)

Reply:
ok, done.